src/common/OpQueue.h

```diff
@@ -64,7 +65,10 @@ class OpQueue {
   virtual void dump(ceph::Formatter *f) const = 0;
 
   // Human readable brief description of queue and relevant parameters
-  virtual void print(std::ostream &f) const = 0;
+  virtual void print(std::ostream &f, const op_queue_type_t &qt) const = 0;
```

Comment:
Why require the caller to pass in the queue type?  The instance already knows its own queue type.  In fact, all of the implementations are identical:

```
    void print(std::ostream &ostream, const op_queue_type_t &qt) const final {
      ostream << get_op_queue_type_name(qt);
    }
```

This is because the behavior of print depends only on the queue enum, which is already exposed from the get_type() interface.  Replace this print() interface with a concrete implementation and remove the implementation from each queue type.

```
    void print(std::ostream &f const {
      ostream << get_op_queue_type_name(get_type());
    }
```

Reply:
Done.